src/index/Permutations.h

```diff
@@ -24,6 +27,23 @@ class PermutationImpl {
         _fileSuffix(std::move(suffix)),
         _keyOrder(order) {}
 
+  // everything that has to be done when reading an index from disk
+  void loadFromDisk(const std::string& onDiskBase) {
+    if constexpr (MetaData::_isMmapBased) {
+      _meta.setup(onDiskBase + ".index" + _fileSuffix + MMAP_FILE_SUFFIX,
+                  ad_utility::ReuseTag(), ad_utility::AccessPattern::Random);
+    }
+    auto filename = string(onDiskBase + ".index" + _fileSuffix);
+    _file.open(filename, "r");
+    AD_CHECK(_file.isOpen());
```

Comment:
I would turn this into an `if` with a proper error message about missing the file and that this means the index is broken.
Since you have this here I think you should remove the `isOpen()` check from the `scans` of course that makes the point about missing that in the two parameter case moot.